src/lib/core/overlay/overlay-directives.ts

```diff
@@ -249,6 +254,12 @@ export class ConnectedOverlayDirective implements OnDestroy {
 
     this._position.withDirection(this.dir);
     this._overlayRef.getState().direction = this.dir;
+    this._closeKeyListener = this._renderer.listen('document', 'keydown',
```

Comment:
This method is getting a bit long. Can you move this new block of code into its own method, maybe `_initEscapeListener()` or something?